### PR TITLE
fix(react): warnings from SideNav in console from inert prop

### LINKED ISSUE #14990: [Bug]: warnings from SideNav in console log from inert prop
### Package

@carbon/react

### Browser

_No response_

### Package version

1.40

### React version

17 and 18 both show the warning

### Description

I believe **_any_** usage of SideNav in one's app will produce this console warning now. Ren Maddox reported this 22 days ago in the Slack carbon-react channel (search for "inert"). Looks like Guilherme reproduced it in StackBlitz, but I don't believe he ever opened an Issue. I have removed the inert prop on the offending line in SideNav.js to keep my log from blowing up with this.

```
Warning: Received `false` for a non-boolean attribute `inert`.

If you want to write it to the DOM, pass a string instead: inert="false" or inert={value.toString()}.

If you used to conditionally omit it with inert={condition && value}, pass inert={condition ? value : undefined} instead.
    at nav
    at SideNavRenderFunction (c:\GitClones\CDD-Hub\node_modules\@carbon\react\lib\components\UIShell\SideNav.js:38:15)
```

### Reproduction/example

see description

### Steps to reproduce

see description

### Suggested Severity

Severity 4 = Unrelated to a user task, has a workaround or does not need a workaround.

### Application/PAL

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by harukikuri**
Closes #14990
### Bug Description

In storybook, **Rail** type of SideNav will produce this console warning.

```
Warning: Received "false" for a non-boolean attribute "inert".

If you want to write it to the DOM, pass a string instead: inert="false" or inert={value.toString()}.

If you used to conditionally omit it with inert={condition && value}, pass inert={condition ? value : undefined} instead.
at nav
at SideNavRenderFunction (c:\GitClones\CDD-Hub\node_modules\@carbon\react\lib\components\UIShell\SideNav.js:38:15)
```

error_msg

### Changelog

**Changed**
- Make sure that it does not pass boolean value to `inert` property for `nav` HTML component. (line.242 in Sidebar.tsx) 

### Testing / Reviewing

**Checklist**
(Cause) 
The `inert` property throwing the error was passed boolean value. 
If a Sidebar component is **Rail** type, this condition returns `false` which `inert` property does not accept. 
It resulted as a console error, only for **Rail** type. 
(Solution) 
When `isRail` property gets passed to SideNav component, that `inert` property should be `undefined`.

**Test** 
After opening up Rail type SideNav component in storybook, there is no this console warning anymore.

### CODE REVIEW

**Comment by kuri-sun** [comment]
I have read the DCO document and I hereby sign the DCO.

**Review by tw15egan — APPROVED** [review]
LGTM once merge is resolved 👍🏻

**Review by andreancardona — APPROVED** [review]
looks good, thank you for the contribution!

**Pull request merged**
